src/diagnostics/radiation_diagnostics.jl

```diff
@@ -942,3 +942,46 @@ add_diagnostic_variable!(
     comments = "In-cloud ratio of the third moment over the second moment of the particle size distribution. Set to zero outside of clouds.",
     compute! = compute_reffcli!,
 )
+
+###
+# Effective radius for ice clouds (3d)
+###
+compute_od550aer!(out, state, cache, time) =
+    compute_od550aer!(out, state, cache, time, cache.atmos.radiation_mode)
+compute_od550aer!(_, _, _, _, radiation_mode::T) where {T} =
+    error_diagnostic_variable("od550aer", radiation_mode)
+
+function compute_od550aer!(
+    out,
+    state,
+    cache,
+    time,
+    radiation_mode::Union{
+        RRTMGPI.AllSkyRadiationWithClearSkyDiagnostics,
+        RRTMGPI.AllSkyRadiation,
+        RRTMGPI.ClearSkyRadiation,
+    },
+)
+    @assert cache.atmos.radiation_mode.aerosol_radiation "aerosol_radiation must be true to enable aerosol optical depth diagnostics"
```

Comment:
@Sbozzolo Is there a better way to do this?

Reply:
We can use the same error system as the other diagnostics by dispatching over Val(cache.atmos.radiation_mode.aerosol_radiation)

Reply:
Let me try

Reply:
I think the alternative is uglier, so I think you can stick with this

Reply:
Ok, thanks!